client/src/main/java/io/pravega/client/ClientFactory.java

```diff
@@ -60,12 +60,22 @@
      *
      * @param scope The scope string.
      * @param controllerUri The URI for controller.
+     * @param creds Optional credentials to connect to controller.
      * @return Instance of ClientFactory implementation.
      */
-    static ClientFactory withScope(String scope, URI controllerUri) {
-        val connectionFactory = new ConnectionFactoryImpl(false);
+    static ClientFactory withScope(String scope, URI controllerUri, PravegaCredentials creds) {
+
+        val connectionFactory = new ConnectionFactoryImpl();
+        boolean enableTls = Boolean.parseBoolean(System.getProperty("io.pravega.auth.enabled"));
+        String tlsCertFile = System.getProperty("io.pravega.auth.certfile");
+
         return new ClientFactoryImpl(scope, new ControllerImpl(controllerUri, ControllerImplConfig.builder().build(),
-                connectionFactory.getInternalExecutor()), connectionFactory);
+                connectionFactory.getInternalExecutor(), creds, enableTls, tlsCertFile), connectionFactory);
```

Comment:
`ConnectionFactoryImpl` already reads this new system properties for its own purpose.   It seems that in this method `ClientFactoruy.withScope`, we only read them to pass them down to `ControllerImpl`.  Maybe `ControllerImpl` can consult these properties itself?